resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverChannelStrategy.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+/**
+ * Strategy that influence how {@link io.netty.channel.socket.DatagramChannel}s are used during queries.
+ */
+public enum DnsNameResolverChannelStrategy {
```

Comment:
@idelpivnitskiy @bryce-anderson @chrisvest I am open for better ideas related to naming...

Reply:
If you ever anticipate reusing this strategy for TCP (especially if we ever make it possible to create a TCP-only resolver), then name lgtm, just adjust the javadoc to avoid mentioning `DatagramChannel`. Imho, we can mention `DatagramChannel` on the builder method instead of this enum. 

If you expect to have 2 different enums for UDP/TCP in the future, then consider renaming this one to `DnsNameResolverDatagramChannelStrategy`.

Reply:
wdyt about `DnsNameResolverSocketStrategy`? The one-to-one relationship between channel and socket is perhaps well known enough for the Netty level, but `Socket` is the actual behavior difference we want to achieve and the fact that it's different channels is just an implementation detail.

For the enum values, how about 

`SocketPerResolver` and `SocketPerResolution`? I like how they're more descriptive but don't like how they're so close and  could be mistaken at a glance so I'm not sold on them myself.

Reply:
PTAL again @bryce-anderson I think I like this name more.